[Steven Wijaya] Duke Increments

stevenwjy opened the pull request.

j-lum committed:
Add support for Gradle workflow

j-lum committed:
Add sample checkstyle configuration

Add toolVersion block in to Gradle code sample to prevent errors.

stevenwjy committed:
Ignore Java class files

stevenwjy committed:
Add code for level 1

stevenwjy committed:
Update read input to be line by line

stevenwjy committed:
Add task for level-2

stevenwjy committed:
Add feature to mark task as done

stevenwjy committed:
Add 3 types of task: ToDo, Event, Deadline

stevenwjy committed:
Add text ui testing

stevenwjy committed:
Fix error handling

stevenwjy committed:
Add feature to remove task

stevenwjy committed:
Fix packages format

stevenwjy committed:
Update InvalidInputException to extend 'Exception' instead of 'RuntimeException'

stevenwjy committed:
Update InvalidInputException implementation

stevenwjy committed:
Rename main package to 'seedu.duke'

stevenwjy committed:
Save task list to a text file and load data from that file at the start of the program.

stevenwjy committed:
Update chatbot to recognise dates and times.

stevenwjy committed:
Merge branch 'branch-Level-7'

stevenwjy committed:
Merge branch 'branch-Level-8'

stevenwjy committed:
Ignore files in the data folder and automatically create empty data/tasks.txt file for task storage

stevenwjy committed:
Update code structure to follow OOP concept more appropiately

stevenwjy committed:
Update runner for text-ui-test and add gitignore for the test output file

stevenwjy committed:
Add template for JUnit Testing

stevenwjy committed:
Merge branch 'gradle'

stevenwjy committed:
Update build.gradle

stevenwjy committed:
Add unit test for ToDo, Event, and Deadline

stevenwjy committed:
Add JavaDoc comments to the code and pass checkstyle evaluation

stevenwjy committed:
Update code formatting and remove switch in TaskType

stevenwjy committed:
Add 'find' command

stevenwjy committed:
Fix response for 'delete' command

stevenwjy committed:
Merge branch 'branch-A-JavaDoc'

stevenwjy committed:
Merge branch 'branch-A-CodingStandard'

stevenwjy committed:
Merge branch 'branch-Level-9'

Fix conflicts in:
- src/main/java/seedu/duke/commands/Command.java
- src/main/java/seedu/duke/io/OutputWriter.java

stevenwjy committed:
Add JavaDocs for findTasks function

stevenwjy committed:
Change unicode characters to 'X' and 'O' to avoid using unicode and pass checkstyle test.

stevenwjy committed:
Add javafx to build.gradle

stevenwjy committed:
Add initial template (layout) for Duke chat app

stevenwjy committed:
Add template for GUI from tutorial

stevenwjy committed:
Refactor code to display output using GUI

stevenwjy committed:
Fix checkstyle

stevenwjy committed:
Change main class name in build.gradle

stevenwjy committed:
Fix task numbering

stevenwjy committed:
Fix design for GUI

stevenwjy committed:
Merge branch 'add-gui'

fangpinsern commented on .gitignore:
```diff
@@ -12,3 +12,13 @@ src/main/resources/docs/
 bin/
+
+# Java class files
+*.class
```
Great idea to ignore the .class files

fangpinsern commented on src/main/java/seedu/duke/commands/Command.java:
```diff
@@ -0,0 +1,70 @@
+            }
+        }
+
+        switch (commandParts[0]) {
```
Great idea to use a switch statement

fangpinsern reviewed (commented):
Overall great code. Just missing javadoc documentation, but i still understand it. So great work!

stevenwjy committed:
Add .travis.yml

stevenwjy committed:
Update .travis.yml

stevenwjy committed:
Permission access for travis

stevenwjy committed:
Add coveralls

stevenwjy committed:
Update setup for coveralls

stevenwjy committed:
Update README.md

stevenwjy committed:
Remove coveralls

stevenwjy committed:
Add assertions for command execution

stevenwjy committed:
Add Assertions

Add assertions for command execution

stevenwjy committed:
Improves Code Quality

stevenwjy committed:
Add function to check equality between task objects

It compares description, type, and additional info (e.g. time description) if there is any

stevenwjy committed:
Prevent adding duplicate tasks

stevenwjy committed:
Add missing javadocs for duplicates detection (#7)

stevenwjy committed:
Add feature to list tasks with '/orderby' parameter

Every task now contains its creation and last updated time.

stevenwjy committed:
Refactor tasks (#9)

* Refactor class naming for different types of task

* Update minor formatting for task package

stevenwjy committed:
Refactor code design (#10)

* Update CommandResult

* Add special response if there is no task for 'list' and 'find' commands

* Add ID to identify tasks uniquely.

* Fix minor formatting

stevenwjy committed:
Update README.md and User Guide

stevenwjy committed:
Add unit tests for commands and tasks

- Test more functionalities of the program
- Update access level for some functions
- Update build.gradle

stevenwjy committed:
Update docs and readme
